toolchain/diagnostics/diagnostic_emitter.h

```diff
@@ -155,6 +181,55 @@ struct DiagnosticBase {
 template <typename LocationT>
 class DiagnosticEmitter {
  public:
+  class DiagnosticBuilder {
+   public:
+    template <typename... Args>
+    auto WithContext(
```

Comment:
Maybe `Note` (the verb) or `NoteContext` (depending on the higher up question) to help remind readers that it will be emitted as a note diagnostic?

Reply:
Done (Note), not sure what you mean by "higher up question".